File: app/src/main/java/com/suhang/networkmvp/ui/activity/MainActivity.java
package com.suhang.networkmvp.ui.activity;

import android.Manifest;
import android.os.Bundle;

import com.suhang.networkmvp.R;
import com.suhang.networkmvp.adapter.MainFragmentAdapter;
import com.suhang.networkmvp.annotation.ActivityScope;
import com.suhang.networkmvp.binding.event.BaseData;
import com.suhang.networkmvp.dagger.module.BlankModule;
import com.suhang.networkmvp.databinding.ActivityMainBinding;
import com.suhang.networkmvp.mvp.model.BlankModel;
import com.suhang.networkmvp.ui.fragment.AttentionFragment;
import com.suhang.networkmvp.ui.fragment.BaseFragment;
import com.suhang.networkmvp.ui.fragment.HomeFragment;
import com.tbruyelle.rxpermissions2.RxPermissions;

import java.util.ArrayList;
import java.util.List;

@ActivityScope
public class MainActivity extends BaseActivity<BlankModel,ActivityMainBinding>{
    private static final String TAG = "MainActivity";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected int bindLayout() {
        return R.layout.activity_main;
    }

    @Override
    protected void subscribeEvent() {

    }

    @Override
    protected void initData() {
        List<BaseFragment> fragments = new ArrayList<>();
        fragments.add(new HomeFragment());
        fragments.add(new AttentionFragment());
        mBinding.vpMain.setAdapter(new MainFragmentAdapter(getSupportFragmentManager(),fragments));
        new RxPermissions(this).request(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE).subscribe(aBoolean -> {
        });
    }

    @Override
    protected BaseData getBindingData() {
        return null;
    }

    @Override
    protected void injectDagger() {
        getBaseComponent().getBlankComponent(new BlankModule()).inject(this);
    }

}

File: app/src/main/java/com/suhang/networkmvp/ui/activity/MainActivity.kt
package com.suhang.networkmvp.ui.activity

import android.Manifest
import android.os.Bundle
import android.support.v4.app.Fragment
import com.suhang.layoutfinderannotation.GenSubComponent
import com.suhang.networkmvp.R
import com.suhang.networkmvp.adapter.MainFragmentAdapter
import com.suhang.networkmvp.annotation.BaseScope
import com.suhang.networkmvp.application.DaggerHelper
import com.suhang.networkmvp.constants.Constant
import com.suhang.networkmvp.dagger.module.BaseModule
import com.suhang.networkmvp.dagger.module.BlankModule
import com.suhang.networkmvp.mvp.model.BlankModel
import com.suhang.networkmvp.ui.fragment.ChildFragment
import com.suhang.networkmvp.ui.fragment.HomeFragment
import com.tbruyelle.rxpermissions2.RxPermissions
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity<BlankModel>() {

    lateinit var adapter: MainFragmentAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bind(R.layout.activity_main)
    }

    override fun subscribeEvent() {

    }

    override fun initData() {
        val fragments = ArrayList<Fragment>()
        fragments.add(HomeFragment())
        fragments.add(ChildFragment())
        adapter = MainFragmentAdapter(supportFragmentManager, fragments)
        vp_main.adapter = adapter
    }

    override fun onDestroy() {
        super.onDestroy()
        adapter.destroy()
    }
}
